lib/publiccloud/azure.pm

```diff
@@ -242,23 +282,23 @@ sub upload_img {
         $file =~ s/\.xz$//;
     }
 
-    my $img_name = $self->get_image_name($file);
-    my $sku = get_var("PUBLIC_CLOUD_AZURE_SKU", $default_sku);
-    my $gen = ($sku =~ "gen2" ? "V2" : "V1");
-    my $os_vhd_uri = $self->get_os_vhd_uri($file);
+    # 1) Create resources
+
+    my $rg_exist = $self->resource_group_exist();
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
-    my $container = $self->container;
+    $self->create_resources($storage_account) if (!$rg_exist);
+
+    # 2) Upload the image blob
 
+    # tags
     my $job_id = get_current_job_id();
     my $openqa_url = get_required_var('OPENQA_URL');
     my $created_by = "$openqa_url/t$job_id";
     my $tags = "openqa_created_by=$created_by";
 
-    my $rg_exist = $self->resource_exist();
-    $self->create_resources($storage_account) if (!$rg_exist);
-
+    my $img_name = $self->get_image_name($file);
+    my $container = $self->container;
```

Comment:
```suggestion
    my $container = $self->container();
```

Reply:
But `$self-> resource_group` is not subroutine.

See how it's defined:
```
lib/publiccloud/azure.pm: 23:has image_gallery => 'test_image_gallery';
lib/publiccloud/azure.pm:21:has resource_group => 'openqa-upload';
```